src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -227,6 +228,18 @@ public void afterPropertiesSet() {
       }
     };
     JavaFxUtil.addListener(fafServerAccessor.connectionStateProperty(), connectionStateInvalidationListener);
+
+    ForgedAlliancePrefs forgedAlliancePrefs = preferencesService.getPreferences().getForgedAlliance();
+    if (preferencesService.isVaultBasePathInvalidForAscii()) {
+      log.warn("Vault base path contains non ASCII characters: {}", forgedAlliancePrefs.getVaultBaseDirectory());
+      if (forgedAlliancePrefs.getWarnNonAsciiVaultPath()) {
+        notificationService.addPersistentWarnNotification("vaultBasePath.nonAscii.warning.title",
+            List.of(new Action(i18n.get("vaultBasePath.nonAscii.warning.ignore"), event -> {
+              forgedAlliancePrefs.setWarnNonAsciiVaultPath(false);
+              preferencesService.storeInBackground();
+            })));
+      }
+    }
```

Comment:
I would move this to an event listener in the PreferencesService that is triggered on a log in event.

Also probably would be better to have a modal dialog like an immediate notification and option to change the path like we do with the game path. As persistent notifications are pretty minor and can be overlooked.

Reply:
Are you sure? A newbie running the client for the first time after logging in will see this dialog instead of getting to know the client?

Reply:
Yes. I think it is most important for people new to the client just like how we ask them to specify the installed game path first thing.

Since it can be a hard bug to track down that can cause launches to fail.